### PR TITLE
Update new contracts to use web3 contract call()

### LINKED ISSUE #1520: Update new smart contract proxies to the new ContractProxy
Implement changes from https://github.com/raiden-network/raiden/pull/1497 in the new `TokenNetwork`, `TokenNetworkRegistry`, `SecretRegistry` proxies.

### PR DESCRIPTION
**Pull request by pcppcp**
Fixes #1520

### CODE REVIEW

**Comment by loredanacirstea** [comment]
@pcppcp , there is also `estimate_and_transact` which existed earlier. This needs to be replaced too.

File: raiden/network/proxies/token_network_registry.py [hunk 1]
```diff
@@ -72,10 +72,8 @@ def token_to_tokennetwork(self, token_address: typing.TokenAddress):
         """ Return the token network address for the given token or None if
         there is no correspoding address.
         """
-        address = self.proxy.call(
-            'token_to_token_networks',
-            token_address,
-        )
+        address = self.proxy.contract.functions.token_to_token_networks(
+            token_address).call()
```
Comment (palango) [inline raiden/network/proxies/token_network_registry.py]: new line after `token_address`

File: raiden/network/proxies/secret_registry.py [hunk 2]
```diff
@@ -65,8 +64,7 @@ def register_secret(self, secret: typing.Secret):
             contract=pex(self.address),
         )
 
-        transaction_hash = estimate_and_transact(
-            self.proxy,
+        transaction_hash = self.proxy.transact(
```
Comment (palango) [inline raiden/network/proxies/secret_registry.py]: `self.proxy.functions.registerSecret(secret).transact()`

Comment (pcppcp) [reply raiden/network/proxies/secret_registry.py]: no, this must be done via the proxy (because of local signatures)

Comment (palango) [reply raiden/network/proxies/secret_registry.py]: Ah, Ok

Comment (loredanacirstea) [reply raiden/network/proxies/secret_registry.py]: So now, all transactions must use `self.proxy.transact`, but calls to query variables can be made through `self.proxy.functions`. Correct?

Comment (pcppcp) [reply raiden/network/proxies/secret_registry.py]: @loredanacirstea for now, yes. But once the `web3` middleware signatures are merged & we decide to switch, we can use standard `web3.eth.functions.f().transact()`.

Comment (loredanacirstea) [reply raiden/network/proxies/secret_registry.py]: Good, because 2 ways of doing transact & call is quite confusing at the moment.

**Pull request merged**